### ISSUE TITLE
A false positive about the rule ClassWithOnlyPrivateConstructorsShouldBeFinal

### ISSUE THREAD

**Issue by Jane-PL-256** [pmd/pmd#4047]
**Affects PMD Version:6.47.0** 

**Rule:ClassWithOnlyPrivateConstructorsShouldBeFinal**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#classwithonlyprivateconstructorsshouldbefinal>

**Description:**
Hi, I found a false positive about this rule, please read the following code example. PMD should not report a warning for this class because `@Value` can add a final modifier for class. Hence, I think this is a false positive. Thanks for your consideration.

**Code Sample demonstrating the issue:**

```java
import lombok.Value;
@Value
public class Bug { // should not report the warning for this class
    private Bug {}
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*

**Comment by jsotuyod** [comment]
This is a duplicate of the already fixed #4200
PMD 6.52.0 and later have properly handled this scenario.
